wails-frontend/src/pages/importPrivateKey.tsx

```diff
@@ -1,97 +1,131 @@
-      return 'Choose an account name';
+  const baselineStr = () =>
+    privateKey ? 'Define your account name' : 'Enter your Private Key';
+
+  const placeholder = () => (privateKey ? 'Account name' : 'Private key');
+
+  async function validateNickname(e: SyntheticEvent) {
+    const formObject = parseForm(e);
+    const { nickname } = formObject;
+
+    if (!nickname) {
+      setError({ nickname: t('nickname-required') });
+      return false;
     }
-    return 'Enter your Private Key';
-  };
 
-  const placeholder = () => {
-    if (privateKey) {
-      return 'Account name';
+    if (await IsAlreadyExists(nickname)) {
+      setError({ nickname: t('nickname-already-exists') });
+      return false;
     }
-    return 'Private key';
-  };
 
-  const handleApply = () => {
-    if (!input || !input.length) {
-      return;
+    const nicknameIsValid = await IsNicknameValid(nickname);
+    if (!nicknameIsValid) {
+      setError({ nickname: t('nickname-invalid-format') });
+      return false;
     }
 
+    return true;
+  }
+
+  async function handleSubmit(e: SyntheticEvent) {
     if (!privateKey) {
-      // TODO: validate private key
+      // if the private key is not defined, set the private key
       setPkey(input);
+      // and reset the input
       setInput('');
-      return;
-    }
-
-    state = { ...state, pkey: privateKey, nickname: input };
+    } else {
+      const nicknameIsValid = await validateNickname(e);
 
-    navigate('/new-password', { state });
-  };
+      // if the private key is defined, validate the nickname
+      if (!nicknameIsValid) return;
+      // if the nickname is valid, navigate to the password page
+      state = { ...state, pkey: privateKey, nickname: input };
+      navigate('/new-password', { state });
+    }
+  }
 
-  const updateInput = (e: React.ChangeEvent<HTMLInputElement>) => {
+  function updateInput(e: React.ChangeEvent<HTMLInputElement>) {
     setInput(e.target.value);
-  };
+  }
 
   return (
     <div className="bg-primary flex flex-col justify-center items-center h-screen w-full">
       <div className="w-1/4  max-w-sm  min-w-fit">
-        <div>
-          <p className="mas-title text-neutral pb-4">{req.Msg}</p>
-        </div>
-        <div>
-          <p className="mas-body text-neutral pb-4">{baselineStr()}</p>
-        </div>
-        {privateKey ? (
-          <div className="pb-4">
-            <Input
-              id="input"
-              onChange={updateInput}
-              autoComplete="off"
-              name="input"
-              placeholder={placeholder()}
-              value={input}
-            />
-          </div>
-        ) : (
-          <div className="pb-4">
-            <Password
-              id="input"
-              onChange={updateInput}
-              autoComplete="off"
-              name="input"
-              placeholder={placeholder()}
-              value={input}
-            />
+        <form ref={form} onSubmit={handleSubmit}>
```

Comment:
I guess we are mixing things here. Controlled and Uncontrolled components. If you want to have all data at once use that onSubmit. But since you are using onChange events I guess you don't need it. 

Am I missing something!? 🤔

Reply:
right, I added form after